python/pyspark/sql/readwriter.py

```diff
@@ -431,6 +432,8 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
         if column is not None:
             if numPartitions is None:
                 numPartitions = self._spark._sc.defaultParallelism
+            assert lowerBound != None, "lowerBound can not be None when ``column`` is specified"
+            assert upperBound != None, "upperBound can not be None when ``column`` is specified"
```

Comment:
Should we resemble the condition here - https://github.com/apache/spark/blob/55d528f2ba0ba689dbb881616d9436dc7958e943/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala#L100-L103 ?